hyperspy/__init__.py

```diff
@@ -39,7 +26,6 @@
 
 import Release
 
+__all__ = ["api"]
```

Comment:
The explanation you give makes sense, and I see why you'd want to add `api`. My only question is if this will hide the other packages from autocomplete in editors/consoles. If not, then I have no issue, if they do, it will be an inconvenience.